### PR TITLE
Avoid sending empty BIOS invetories to Hollow

### PR DESCRIPTION
**Pull request by ScottGarman**
The 'packet-hardware inventorybios' command can succeed but generate
empty BIOS JSON on unsupported platforms. Check for this and avoid
writing to Hollow in these cases.

The `$id` that we get from metadata is the instance uuid, not the hardware uuid. Make the hwuuid available to osie scripts by passing it down via osie runner's docker env.

Also fixed a typo in the hollow namespace.

### CODE REVIEW

File: docker/scripts/functions.sh [hunk 1]
```diff
@@ -403,23 +403,31 @@ function validate_bios_config() {
...
+	local plan=$1
+	local facility=$2
 
 	# When running the inventorybios command outside of the packet-hardware
 	# container, UTIL_RACADM7 and UTIL_SUM must be set to the locations of the
 	# racadm and sum binaries, respectively
 	local bios_json_fn="/tmp/bios.json"
 	local hollow_json_fn="/tmp/hollow.json"
-	local hollow_namespace="net.equinixplatform.bios"
+	local hollow_namespace="net.platformequinix.bios"
```
Comment (tstromberg) [inline docker/scripts/functions.sh:417]: Please mention this change in the PR description and/or move it into another PR.

File: docker/scripts/functions.sh [hunk 2]
```diff
@@ -403,23 +403,31 @@ function validate_bios_config() {
...
 	if UTIL_RACADM7=/opt/dell/srvadmin/bin/idracadm7 UTIL_SUM=/opt/supermicro/sum/sum packet-hardware inventorybios --verbose -u localhost --dry --cache-file "${bios_json_fn}"; then
+		local inventorybios_json
+		inventorybios_json=$(cat ${bios_json_fn})
```
Comment (tstromberg) [inline docker/scripts/functions.sh]: Always quote command substitutions (unless you need unquoted behavior): https://google.github.io/styleguide/shellguide.html#quoting

File: docker/scripts/functions.sh [hunk 3]
```diff
@@ -403,23 +403,31 @@ function validate_bios_config() {
...
+		echo "inventorybios JSON is: [${inventorybios_json}]"
+
+		if [[ ${inventorybios_json} == "{}" ]]; then
```
Comment (tstromberg) [inline docker/scripts/functions.sh:423]: you should add quotes here too, otherwise certain strings ("", "|") may act funny.

File: docker/scripts/functions.sh [hunk 4]
```diff
@@ -403,23 +403,31 @@ function validate_bios_config() {
...
+			echo "inventorybios JSON is empty, not writing to Hollow"
+			return 0
+		fi
+
 		# Generate JSON with additional fields required by Hollow
 		if ! jq --null-input \
 			--arg ns "${hollow_namespace}" \
-			--arg id "${id}" \
+			--arg id "${HARDWARE_ID}" \
```
Comment (tstromberg) [inline docker/scripts/functions.sh]: It feels really weird to pass this in as a global environment variable, when the other inputs are arguments. I'd recommend consistency. 

If you do need to rely on global variables passed-in from elsewhere - please document them as as a doc string in the function signature.

**Review by tstromberg — CHANGES_REQUESTED** [review]

File: docker/scripts/functions.sh (hunk 1 above)
Comment (ScottGarman) [reply docker/scripts/functions.sh:417]: This change was done in a separate commit, but I failed to mention it in the PR description. That's been fixed, and I've improved the PR description to be more complete in other ways, too.

File: docker/scripts/functions.sh (hunk 2 above)
Comment (ScottGarman) [reply docker/scripts/functions.sh]: Fixed.

File: docker/scripts/functions.sh (hunk 3 above)
Comment (ScottGarman) [reply docker/scripts/functions.sh:423]: Fixed (I keep thinking shellcheck will catch these, but it doesn't).

File: docker/scripts/functions.sh (hunk 4 above)
Comment (ScottGarman) [reply docker/scripts/functions.sh]: The reason for this change wasn't mentioned in the commit message or PR description - I've updated both for clarity as to why. I agree that it's clearer to pass the value is a function parameter, so I've made that change as well.

File: docker/scripts/functions.sh (hunk 3 above)
Comment (ScottGarman) [reply docker/scripts/functions.sh:423]: Actually, I misspoke. It's `shfmt` which doesn't like those quotes, so I have to leave them off in order to pass CI. This doesn't seem right to me either, and a quick search online doesn't reveal the reasoning behind this - I may have to dig in further.

Comment (mmlb) [reply docker/scripts/functions.sh:423]: the quotes issues is *only* for `sh`'s `[` and not for `bash`'s `[[ ]]` which is builtin:

```ShellSession
14:52  ~
$ foo='"'

14:52  ~
$ [[ $foo == '' ]] && echo true

14:52  ~
$ [[ $foo == '|' ]] && echo true

14:52  ~
$ [[ $foo == '"' ]] && echo true
true

14:52  ~
$ foo='|'

14:52  ~
$ [[ $foo == '' ]] && echo true

14:52  ~
$ [[ $foo == '|' ]] && echo true
true

14:52  ~
$ [[ $foo == '"' ]] && echo true
```

**Review by tstromberg — APPROVED** [review]
